core/src/main/java/io/fabric8/maven/core/service/kubernetes/JibBuildService.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.service.kubernetes;
+
+import io.fabric8.maven.core.service.BuildService;
+import io.fabric8.maven.docker.config.BuildImageConfiguration;
+import io.fabric8.maven.docker.config.ImageConfiguration;
+import io.fabric8.maven.docker.util.ImageName;
+import io.fabric8.maven.docker.util.Logger;
+
+import java.util.List;
+import java.util.Objects;
+
+public class JibBuildService implements BuildService {
+
+    private BuildServiceConfig config;
+
+    private Logger log;
+    private JibBuildService() { }
+
+    public JibBuildService (BuildServiceConfig config, Logger log) {
+        Objects.requireNonNull(config, "config");
```

Comment:
I would like the error message to be a bit more informative

Reply:
The constructor is visible for testing purpose